sklearn/tree/tests/test_tree.py

```diff
@@ -1693,17 +1693,28 @@ def test_no_sparse_y_support(name):
 
 
 def test_mae():
-    # check MAE criterion produces correct results
+    # Check MAE criterion produces correct results
     # on small toy dataset
     dt_mae = DecisionTreeRegressor(random_state=0, criterion="mae",
                                    max_leaf_nodes=2)
+
+    # Test MAE where all sample weights are uniform:
+    dt_mae.fit([[3], [5], [3], [8], [5]], [6, 7, 3, 4, 3], np.ones(5))
+    assert_array_equal(dt_mae.tree_.impurity, [1.4, 1.5, 4.0 / 3.0])
+    assert_array_equal(dt_mae.tree_.value.flat, [4, 4.5, 4.0])
```

Comment:
How do we know that these are the right values. It would be useful to explain it in a comment.

Reply:
I explained in the 2nd post above... Perhaps I could put a link to this pull in the comments? Otherwise I can try to summarise?